align/pdk/finfet/basic_template.sp

```diff
@@ -288,7 +302,7 @@ M2 DB DA SB B NMOS w=w l=90n
 .ends LS_NMOS_B
 
 .subckt LS_S_PMOS_B DA DB SA SB B
-* @: Generator(parameters={'pattern':'ncc'})
+* @: Generator(name='mos', parameters={'pattern':'ncc'})
```

Comment:
@kkunal1408 , 
why is name='mos' required in this PR? 
what does pattern='ncc' mean?

Reply:
'mos' is to annotate that this primitive will be generated using MOS-generator
'ncc' is non-common-centroid-@arvuce22 suggested that primitives which do not have common source connections are better using con common centroid. So, I kept it as a placeholder in the constraint so that he can use it to modify the MOS-generator.

Reply:
Is 'mos' hard-coded to load 'MOSGenerator' then?
'ncc' is confusing. I will revise this file shortly. I would suggest to exclude the pattern attribute or set it to None for other PDKs.

Reply:
yes 'mos' is hardcoded to MOSGenerator in the primitive generator. Previously it was based on string match 'MOS' in the template_name.

Reply:
Sure you can modify 'ncc' as this is just a placeholder right now.

Reply:
> yes 'mos' is hardcoded to MOSGenerator in the primitive generator. Previously it was based on string match 'MOS' in the template_name.

I missed syncing this file. Just updated primitive/mani.py (line 162-168)

Reply:
> Sure you can modify 'ncc' as this is just a placeholder right now.

This is supposed to replace this hardocded line (line 46) in primitive/main.py 
shared_diff = 0 if any(primitive.name.startswith(f'{x}_') for x in ["LS_S", "CMC_S", "CCP_S"]) else 1